### PR TITLE
Normalize a few more spin-wait loops

### PR DESCRIPTION
**Pull request by kouvel**
- Fixed a few more spin-waits to normalize the spin-wait duration between processors
- These spin-waits have so far not needed to be retuned to avoid unreasonably long spin-wait durations. They can be retuned as necessary in the future.
  - Added a version of YieldProcessorNormalized() that normalizes based on spin-wait counts tuned for pre-Skylake processors for spin-wait loops that have not been retuned.
- Moved some files around to make YieldProcessorNormalized() and the like available in more places. Initialization is still only done in the VM. Uses outside the VM will use the defaults, where there would be no significant change from before.

### CODE REVIEW

**Comment by kouvel** [comment]
@dotnet-bot test Windows_NT x64 Formatting

**Comment by vancem** [comment]
@kouvel  - If I understand this right, YieldProcessorNormalizedForPreSkylakeCount is basically a marker that we have not thought about the yield count much, and thus arguably could be wrong.   Ideally all of these uses go away eventually.   In the mean time, we assume that the count should be divided by 8 to get the 'normalized' yield count.  

Also I am assuming that we want all YieldProcessor() calls gone (does this change complete that mission?  If so we should make it prviate).   

It would be good if in general we put some good comments in yieldProcessorNormalzied.h,  Conceptually what is going on is not hard (we want a yield that represents a certain amount of time) regardless of processor, and that Intel processors after Skylake have a 'yield' instruction that is much slower than processors before that.      THis is hard to discover from code but easy to simply state in the YieldProcessorNormalized.h file.  

In particular we should have a comment on  YieldProcessorNormalizedForPreSkylakeCount  that tells people the reasoning behind the method and how it is to be used (and hopefully ultimately removed).  

But I don't have a problem with the actual delta.

File: src/inc/yieldprocessornormalized.h [hunk 1]
```diff
@@ -69,6 +78,50 @@ FORCEINLINE void YieldProcessorNormalized(const YieldProcessorNormalizationInfo
...
+            preSkylakeCount = MaxCount;
+        }
+    }
+
+    const unsigned int PreSkylakeCountToSkylakeCountDivisor = 8;
+    SIZE_T n = (SIZE_T)preSkylakeCount * normalizationInfo.yieldsPerNormalizedYield / PreSkylakeCountToSkylakeCountDivisor;
+    if (n == 0)
+    {
+        n = 1;
+    }
```
[the remark this answers is not in the document]
Comment (kouvel) [reply src/inc/yieldprocessornormalized.h]: The value is currently unsigned

**Comment by kouvel** [comment]
> Were these changes speculative

> If I understand this right, YieldProcessorNormalizedForPreSkylakeCount is basically a marker that we have not thought about the yield count much

Yea I haven't seen many reports of issues in these spin-waits (maybe one minor case that I sort of recall). They are not necessarily short spin-waits, so there is potential for it to become an issue. Now that the yield normalization stuff has been in for a while and seems to be working as expected, I'm just normalizing the remaining spin-waits to avoid bad hardware-specific scaling. Retuning them can be done later if/when necessary, and after all of those cases are done the function would not be necessary anymore.

> Also I am assuming that we want all YieldProcessor() calls gone (does this change complete that mission? If so we should make it prviate).

The only other uses are in the GC, where scaling is done once on a unit value that is used elsewhere, and the code has to be compatible with NetCore and NetFx. I have now made YieldProcessor() private outside the GC.

Addressed other feedback as well.

**Pull request merged**